pkg/curatedpackages/reader.go

```diff
@@ -0,0 +1,87 @@
+package curatedpackages
+
+import (
+	"context"
+	"fmt"
+
+	eksdv1 "github.com/aws/eks-distro-build-tooling/release/api/v1alpha1"
+	"sigs.k8s.io/yaml"
+
+	packagesv1 "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/manifests"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+type PackageReader struct {
+	ManifestReader *manifests.Reader
+}
+
+func NewPackageReader(mr *manifests.Reader) *PackageReader {
+	return &PackageReader{
+		ManifestReader: mr,
+	}
+}
+
+func (r *PackageReader) ReadBundlesForVersion(version string) (*releasev1.Bundles, error) {
+	return r.ManifestReader.ReadBundlesForVersion(version)
+}
+
+func (r *PackageReader) ReadEKSD(eksaVersion, kubeVersion string) (*eksdv1.Release, error) {
+	return r.ManifestReader.ReadEKSD(eksaVersion, kubeVersion)
+}
+
+func (r *PackageReader) ReadImages(eksaVersion string) ([]releasev1.Image, error) {
+	return r.ManifestReader.ReadImages(eksaVersion)
+}
+
+func (r *PackageReader) ReadImagesFromBundles(b *releasev1.Bundles) ([]releasev1.Image, error) {
+	images, err := r.ManifestReader.ReadImagesFromBundles(b)
+	for _, v := range b.Spec.VersionsBundles {
+		images = append(images, v.PackageControllerImage()...)
+	}
+	return images, err
+}
+
+func (r *PackageReader) ReadCharts(eksaVersion string) ([]releasev1.Image, error) {
+	return r.ManifestReader.ReadCharts(eksaVersion)
+}
+
+func (r *PackageReader) ReadChartsFromBundles(ctx context.Context, b *releasev1.Bundles) []releasev1.Image {
+	images := r.ManifestReader.ReadChartsFromBundles(ctx, b)
+	for _, vb := range b.Spec.VersionsBundles {
+		artifact := GetPackageBundleRef(vb)
+		packages, err := fetchPackages(ctx, vb, artifact)
+		if err != nil {
+			logger.MarkFail("error finding packages for artifact", artifact, "error", err)
```

Comment:
shouldn't this return an error?
At the end if the user wants all the charts exported and we are not able to pull one, shouldn't we error out?

Reply:
Ideally not. We might not have packagebundle manifest for every kubeversion that is currently supported. We only fetch those we can get.

Reply:
Discussed offline, we will keep as is but revisit the error handling before GA.